data/routers/usersRouter.js

```diff
@@ -41,9 +55,12 @@ router.post('/login', async (req, res) => {
   if (body && body.username && body.password) {
     const user = await Users.findUser(body)
     if (!user || !bcrypt.compareSync(body.password, user.password)) {
-      return res.status(401).json({ error: 'You shall not pass!' })
+      if (req.session && !req.session.user) {
+        return res.status(401).json({ error: 'You shall not pass!' })
+      }
```

Comment:
Typically you'd move this logic into a helper function, which usually is used as middleware:
```javascript
const protectSession = (req, res, next) => {
  if (req.session && req.session.user) {
    next();
  } else {
    res.status(401).json({ message: 'You shall not pass!' });
  }
};
```
which you could pass as middleware directly in the applicable routes:
```javascript
router.get('/', protectSession, async (req, res) => {
  // GET logic
});
```

Reply:
It also looks like there could be some bad happenings with this condition:
```javascript
req.session && !req.session.user
```
I think we want this to be more specific; currently, it's asking:
`is there a session and is there not a user on that session?`
if yes to both, then throw an error.
What we want to know firstly is there's a session and if there's a user.
If so, then we give access to the route.
If not, then we throw an error.

Reply:
Specifically, imagine this check on a route to the list of users. What happens when the client has no session? They get to see the list of users.